pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -28,6 +28,43 @@
 INV_SQRT2 = 1 / qml.math.sqrt(2)
 
 
+class Barrier(Operation):
+    r"""Barrier(wires)
+    The Barrier operator, used to separate the compilation process into blocks or as a visual tool.
+
+    **Details:**
+
+    * Number of wires: AnyWires
+    * Number of parameters: 0
+
+    Args:
+        only_visual (bool): True if we do not want it to have an impact on the compilation process. Default is False.
+        wires (Sequence[int] or int): the wires the operation acts on
+    """
+    num_params = 0
+    num_wires = AnyWires
+    par_domain = None
+
+    def __init__(self, only_visual=False, *params, wires=None, do_queue=True, id=None):
+
```

Comment:
We should be able to broadcast `qml.Barrier` across all available wires, with simply just calling `qml.Barrier()`.

`qml.state()` and `qml.probs()`, the two objects that can currently broadcast across all wires, indicating broadcasting across all wires by an empty wires object `Wires([])`.  I do think this behaviour should be improved upon in the future, but for now that's how we're denoting "on all wires".

Reply:
I think at one point we considered using `qml.operation.AllWires`, but decided it is too much overhead for users 🤔  Enums are nice for developers, but not so nice for user arguments (hence why SciPy uses strings everywhere)

Reply:
I was thinking of creating class `AllWires` that inherits from `Wires` and defines special properties and arithmetics.